rgw/admin/caps.go

```diff
@@ -0,0 +1,56 @@
+package admin
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+// AddUserCap will add the capabilities for a user
+func (api *API) AddUserCap(ctx context.Context, uid, userCap string) ([]UserCapSpec, error) {
+	if uid == "" {
+		return nil, errMissingUserID
+	}
+	if userCap == "" {
+		return nil, errMissingUserCap
+	}
+
+	user := User{ID: uid, UserCaps: userCap}
+	body, err := api.call(ctx, http.MethodPut, "/user?caps", valueToURLParams(user))
+	if err != nil {
+		return nil, err
+	}
+
+	var ref []UserCapSpec
+	err = json.Unmarshal(body, &ref)
+	if err != nil {
+		return nil, fmt.Errorf("%s. %s. %w", unmarshalError, string(body), err)
+	}
+
+	return ref, nil
+}
+
+// RemoveUserCap will remove the capabilities from a user
+func (api *API) RemoveUserCap(ctx context.Context, uid, userCap string) ([]UserCapSpec, error) {
```

Comment:
So this removes the specified caps and returns the list of remaining caps?

Reply:
Yup

Reply:
So should we call this Update from an API point of view? And have a `RemoveUserCap` that should remove the caps but returns nothing? It feels strange to remove something but return it. For instance, if we remove a User we don't return anything.
What do you think?

Reply:
@leseb  I followed semantics from radosgw-admin cli.  IMO both add and remove caps API does kind of the update functionality, so I am not sure renaming remove to update. Or we can Update API() which does both add/remove but not simultaneously?

Reply:
Could this be embedded as part of the `ModifyUser()` maybe?

Reply:
I guess the end-user may expect separate APIs for caps, both in AdminOps or radosgw-admin it is not handled in modifyUser or equivalent call. Secondly, it makes more calls to ModifyUser API and additional fields to User struct{}(if the signature is not planned to change). I  really like the current approach of Api one type of HTTP request and handle its response

Reply:
If I am understanding the last part of this exchange correctly I would not like to see one api embedded within another. One aspect of API design that I like to follow is to do one thing whenever possible. So, if this is a unique endpoint in the rgw http scheme then I agree with having a unique function for it as well.

 Personally I have no issue with the call returning the new capabilities set after removing the specified caps  (assuming that's what it does) - I would recommend documenting that in the doc comment though! :-)

Reply:
In general with go-ceph as a binding library we follow the guideline to make the binding layer as thin as possible. We only leave that path for one reason, that is to make the API more go-idiomatic. Since this is not the issue here, I would clearly vote to stay as close as possible to the Rest-API, which is in fact returning the updated list of capabilities. Beside that fact, I also don't see a problem that a delete function returns the updated list, which is very common in functional languages where you deal with immutable data structures. So if you add a return value description in the comment (which we should do in general), for me everything is green here.

Reply:
There is still a comment missing, right?

Reply:
@ansiwen : I have added the following comment before return values at L:35 and L:60
```
// returns updated list of UserCaps for the user
```
Am I missing anything ?sorry if I misterstood it completely

Reply:
Oh, yes, I guess you misunderstood. We need that comment in the comment of the function, so that it will be added to the documentation and therefore visible for the users.

Reply:
Before the "PREVIEW" line, please.

Reply:
@ansiwen please use the suggestion feature so we can sort it out.